File: src/client/java/fr/hardel/asset_editor/client/javafx/lib/data/SlotConfigs.java
package fr.hardel.asset_editor.client.javafx.lib.data;

import net.minecraft.resources.Identifier;

import java.util.List;
import java.util.Map;
import java.util.function.Function;
import java.util.stream.Collectors;

public final class SlotConfigs {

    public record SlotConfig(String id, List<String> slots) {

        public Identifier image() {
            return SlotConfigs.slotImage(id);
        }
    }

    public static final List<SlotConfig> ALL = List.of(
        new SlotConfig("mainhand", List.of("mainhand", "any", "hand", "all")),
        new SlotConfig("offhand",  List.of("offhand", "any", "hand", "all")),
        new SlotConfig("body",     List.of("body", "any", "all")),
        new SlotConfig("saddle",   List.of("saddle", "any", "all")),
        new SlotConfig("head",     List.of("head", "any", "armor", "all")),
        new SlotConfig("chest",    List.of("chest", "any", "armor", "all")),
        new SlotConfig("legs",     List.of("legs", "any", "armor", "all")),
        new SlotConfig("feet",     List.of("feet", "any", "armor", "all"))
    );

    public static final Map<String, SlotConfig> BY_ID = ALL.stream()
            .collect(Collectors.toUnmodifiableMap(SlotConfig::id, Function.identity()));

    private static Identifier slotImage(String name) {
        return Identifier.fromNamespaceAndPath("minecraft", "textures/studio/slots/" + name + ".png");
    }

    private SlotConfigs() {}
}

File: src/client/kotlin/fr/hardel/asset_editor/client/compose/lib/data/SlotConfigs.kt
package fr.hardel.asset_editor.client.compose.lib.data

import net.minecraft.resources.Identifier

object SlotConfigs {

    data class SlotConfig(val id: String, val slots: List<String>) {
        fun image(): Identifier = slotImage(id)
    }

    @JvmField
    val ALL = listOf(
        SlotConfig("mainhand", listOf("mainhand", "any", "hand", "all")),
        SlotConfig("offhand", listOf("offhand", "any", "hand", "all")),
        SlotConfig("body", listOf("body", "any", "all")),
        SlotConfig("saddle", listOf("saddle", "any", "all")),
        SlotConfig("head", listOf("head", "any", "armor", "all")),
        SlotConfig("chest", listOf("chest", "any", "armor", "all")),
        SlotConfig("legs", listOf("legs", "any", "armor", "all")),
        SlotConfig("feet", listOf("feet", "any", "armor", "all"))
    )

    @JvmField
    val GROUPS = listOf(
        listOf("mainhand", "offhand"),
        listOf("body", "saddle"),
        listOf("head", "chest", "legs", "feet")
    )

    @JvmField
    val BY_ID = ALL.associateBy(SlotConfig::id)

    private fun slotImage(name: String): Identifier =
        Identifier.fromNamespaceAndPath("minecraft", "textures/studio/slots/$name.png")
}
